App/BitBar/ExecutablePlugin.m

```diff
@@ -70,86 +70,97 @@ -(void)performRefreshNow:(NSMenuItem*)menuItem{
 }
 
 -(BOOL)refresh {
+  __unsafe_unretained ExecutablePlugin *weakSelf = self;
   [self.lineCycleTimer invalidate];
   self.lineCycleTimer = nil;
   [self.refreshTimer invalidate];
   self.refreshTimer = nil;
-    
+
   // execute command
   dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT,0),  ^{
-    [self refreshContentByExecutingCommand];
+    [weakSelf refreshContentByExecutingCommand];
     dispatch_sync(dispatch_get_main_queue(), ^{
-      
-      self.lastUpdated = NSDate.new;
-      
-      [self rebuildMenuForStatusItem:self.statusItem];
-      
-      // reset the current line
-      self.currentLine = -1;
-      
-      // update the status item
-      [self cycleLines];
-      
-      // sort out multi-line cycler
-      if (self.isMultiline) {
+      if (weakSelf) {
+        __strong ExecutablePlugin* strongSelf = weakSelf;
```

Comment:
Originally I didn't do that as I failed to notice that the second call was `dispatch_sync` and didn't want to create Yet Another Circular Reference.  But give it is `sync` it should be fine.  I ran a quick test, seems to be okay, push another commit.